File: tasks/model-validation/validate.R
library(here)
library(yaml)

# Load YAML config
config <- yaml.load_file(here("config", "config.yaml"))

cat("Raw data path:", config$data$raw, "\n")

df <- read.csv(here(config$data$raw))
mse <- mean((df$y - df$x*2)^2)
print(paste('MSE:', mse))

File: tasks/model-validation/validate.py
import pandas as pd
from sklearn.metrics import mean_squared_error
from pathlib import Path
import yaml

config_path = Path(__file__).resolve().parents[2] / "config" / "config.yaml"
config = yaml.safe_load(config_path.read_text())

print("Raw data path:", config["data"]["raw"])

df = pd.read_csv(config["data"]["raw"])
preds = df['x'] * 2
print('MSE:', mean_squared_error(df['y'], preds))
